### ISSUE TITLE
Enable n_warmup in all mcmc functions

### ISSUE THREAD

**Issue by wds15** [stan-dev/bayesplot#76]
Problem:

Users extract all draws from a stan object and then would like to use the same draws with all mcmc functions. However, at the moment the user has to extract all draws and a draws object which does not include the warmup iterations as only the mcmc_trace function understands the n_warmup argument.

While for the mcmc_trace function the plot actually changes by shading the warmup area, all other mcmc functions should also allow for the n_warmup argument in order to avoid user mistakes as users tend to use a single export of the draws. For most mcmc functions the n_warmup argument only leads to subsetting the draws, nothing more.

**Comment by jgabry** [comment]
I do like the simplicity of assuming the user has removed warmup draws, but
I see the utility of adding the argument to all the mcmc functions.

Out of curiosity, why is your default to include warmup when extracting the
draws? Wouldn't it be preferable to only bother to extract with warmup if
there's anything fishy post-warmup? If there isn't then you can just
proceed with the post-warmup draws. Anyway, that's the workflow I had in
mind, but it's not necessarily the best.

**Comment by wds15** [comment]
Maybe you are right. I always thought that showing the whole thing and starting iterations from 0 is what users expect. But I agree, that warmup draws are possibly not worth to look at ever (honestly, users ask for traceplots - I have stopped looking at them once I started to use Stan and only look at them when things go wrong which is not so often).

This feature should have low prio and maybe just be addressed with some extra doc somewhere to not let users run into trouble.
